dace/codegen/targets/rtl.py

```diff
@@ -0,0 +1,519 @@
+        self.code_objects.append(CodeObject(name="{}".format(unique_name),
+                                            code=RTLCodeGen.rtl_header().format(name=unique_name,
+                                                                                parameters=parameter_string,
+                                                                                inputs="\n".join(inputs),
+                                                                                outputs="\n".join(outputs))
+                                                 + tasklet.code.code
+                                                 + RTLCodeGen.rtl_footer(),
+                                            language="sv",
+                                            target=RTLCodeGen,
+                                            title="rtl",
+                                            target_type="",
+                                            additional_compiler_kwargs="",
+                                            linkable=True,
+                                            environments=None))
+
+        # generate verilator simulation cpp code components
+        inputs, outputs = self.generate_cpp_inputs_outputs(tasklet)
+        vector_init = self.generate_cpp_vector_init(tasklet)
+        num_elements = self.generate_cpp_num_elements()
+        internal_state_str, internal_state_var = self.generate_cpp_internal_state(tasklet)
+
+        # add header code do stream
+        sdfg.append_global_code(cpp_code=RTLCodeGen.header_template().format(name=unique_name))
+
+        # add main cpp code to stream
+        callsite_stream.write(contents=RTLCodeGen.main_template().format(name=unique_name,
+                                                                         inputs=inputs,
+                                                                         outputs=outputs,
+                                                                         num_elements=num_elements,
+                                                                         vector_init=vector_init,
+                                                                         internal_state_str=internal_state_str,
+                                                                         internal_state_var=internal_state_var),
+                              sdfg=sdfg,
+                              state_id=state_id,
+                              node_id=node)
+
+    @staticmethod
+    def header_template():
+        """
+            Cpp Header Template
+        """
+        return """\
+// generic includes
+#include <iostream>
+
+// verilator includes
+#include <verilated.h>
+
+// include model header, generated from verilating the sv design
+#include "V{name}.h"
+"""
+
+    @staticmethod
+    def main_template():
+        """
+            Cpp Main Verilator Simulator Template
+        """
+        return """\
+std::cout << "SIM START" << std::endl;
```

Comment:
Be more specific about what is being simulated. If multiple tasklets are running simultaneously, this might get messy...